lldb/source/Plugins/SymbolLocator/Debuginfod/SymbolLocatorDebuginfod.cpp

```diff
@@ -133,10 +213,32 @@ static std::optional<FileSpec> GetFileForModule(
 
 std::optional<ModuleSpec> SymbolLocatorDebuginfod::LocateExecutableObjectFile(
     const ModuleSpec &module_spec) {
-  return GetFileForModule(module_spec, llvm::getCachedOrDownloadExecutable);
+  switch (GetGlobalPluginProperties().GetLookupMode()) {
+  case eLookupModeAlways:
+    return GetFileForModule(module_spec, llvm::getCachedOrDownloadExecutable);
+  /*
+  case eLookupModeOnDemand:
+    // TODO
+  case eLookupModeAsync:
+    // TODO
+  */
+  case eLookupModeNever:
+    return {};
+  }
 }
 
 std::optional<FileSpec> SymbolLocatorDebuginfod::LocateExecutableSymbolFile(
     const ModuleSpec &module_spec, const FileSpecList &default_search_paths) {
```

Comment:
add `bool user_initiated` as a new parameter.

Reply:
After further investigation, there's no need: force_lookup is already "user initiated" or "background lookup" which is sufficient.